src/test/resources/processor_test_files/1118_UtilityClassWithPublicConstructor/UtilityClassWithImplicitPublicConstructor.java.expected

```diff
@@ -0,0 +1,9 @@
+public class UtilityClassWithImplicitPublicConstructor { // Noncompliant
+    private UtilityClassWithImplicitPublicConstructor() {}
```

Comment:
I don't understand. Do you mean that there's a `// Noncompliant` comment in there?

Reply:
Ohhhh, wait, now I understand, the constructor throws an exception. Not sure why I didn't get it.

I'm not sure if we want to add that. It will decrease coverage metrics as it adds a line of code that we then don't test, which might be an annoyance to some projects. I don't really think the exception is necessary, as a) the private constructor can only be used inside of the utility class itself, and b) it's not actually a big problem if the class is instantiated, it's just weird and unnecessary.

Reply:
Fine! Then, let's merge the PR.